baker/GrapherBaker.tsx

```diff
@@ -366,29 +366,48 @@ const bakeAllPublishedChartsVariableDataAndMetadata = async (
         }
     )
 
-    const maxWorkers = MAX_NUM_BAKE_PROCESSES
-    const poolOptions = {
-        minWorkers: 2,
-        maxWorkers: maxWorkers,
-    }
-    const pool = workerpool.pool(__dirname + "/worker.js", poolOptions)
     const jobs: BakeVariableDataArguments[] = variableIds.map((variableId) => ({
         bakedSiteDir,
         variableId,
         checksumsDir,
     }))
-    try {
+
+    if (MAX_NUM_BAKE_PROCESSES == 1) {
         await Promise.all(
-            jobs.map((job) =>
-                pool.exec("bakeVariableData", [job]).then((job) =>
-                    progressBar.tick({
-                        name: `variableid ${job.variableId}`,
-                    })
+            jobs.map(async (job) => {
+                await bakeVariableData(job)
+                progressBar.tick({ name: `variableid ${job.variableId}` })
+            })
+        )
+    } else {
+        const maxWorkers = MAX_NUM_BAKE_PROCESSES
+        const poolOptions = {
+            minWorkers: 2,
+            maxWorkers: maxWorkers,
+            // using `process` instead of worker threads is necessary for DuckDB to work
+            workerType: "process",
+        } as workerpool.WorkerPoolOptions
+        const pool = workerpool.pool(__dirname + "/worker.js", poolOptions)
```

Comment:
This is fine, but just fyi, `path` exists for node scripts if you're wanting safely to work with paths without having to worry about accidentally writing `foo//bar`

Reply:
Any idea if there's a library that would support both URLs and file paths? (`path.join` converts `//` into `/` in URLs) Just curious, I had to do it with string concat.